### PR TITLE
Oauth test

### PR DESCRIPTION
**Pull request by arnaldoaparicio**
# Description

Please check the type of change your PR introduces:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

Allow user creation using Google OAuth2.

 

### Type of change

Please check the relevant option.

- [x] Non-Breaking Change
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Big Breaking change (this breaks everything and I am sorry)

# How Has This Been Tested?

Please check the option(s) that best describe your tests. 

- [ ] New spec files were added
- [ ] Tests were added to existing spec files
- [ ] Some tests were changed
- [x] No tests were changed
- [ ] All tests were changed (please explain what in God's name happened)

### Tests Description

- Please add further details of your tests here.

# Checklist:

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have removed all unused code or commented why I left it
- [ ] I have deployed my changes to a test-heroku and they work
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass locally with my changes

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    auth_hash = request.env['omniauth.auth']
+    email = auth_hash['info']['email']
+
+    session[:access_token] = auth_hash['credentials']['token']
+
+    response = conn.post('/api/v1/users') do |req|
+      req.headers['Content-Type'] = 'application/json'
+      req.body = JSON.generate(user: { username: email, email: email })
+    end
+    redirect_to '/'
```
Comment (jbreit88) [inline app/controllers/users_controller.rb:14]: This should probably redirect to dashboard

File: app/poros/book.rb [hunk 2]
```diff
@@ -7,12 +7,12 @@ class Book
...
-    @book_cover = data[:book_cover]
-    @rating = data[:rating]
-    @genre = data[:genre]
+    @title = data[:data][:attributes][:title]
+    @author = data[:data][:attributes][:authors][0]
+    @summary = data[:data][:attributes][:description]
+    # @reviews = data[:data][:attributes][:reviews]
+    # @book_cover = data[:book_cover]
+    @rating = data[:data][:attributes][:rating]
+    @genre = data[:data][:attributes][:genres][0]
```
Comment (jbreit88) [inline app/poros/book.rb:16]: Need to add a guard statement in case genre is nil.

**Review by jbreit88 — APPROVED** [review]
I have left some comments about stuff that needs to be fixed or updated, but I don't see anything that is malfunctioning or broken, just more functionality we need to add on in the future. Looks good to merge.

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (Sierra-T-9598) [reply app/controllers/users_controller.rb:14]: Agreed

**Review by Sierra-T-9598 — APPROVED** [review]

**Review by Malllll12 — APPROVED** [review]

**Pull request merged**